spec/spec_helper.rb

```diff
@@ -14,3 +14,13 @@
     c.syntax = :expect
   end
 end
+
+if ENV["CI"] == "true" && /^\d+\.\d+/.match(RUBY_VERSION)[0] == "2.7"
```

Comment:
@jemmaissroff in order to keep the Ruby version logic in one place -- just `main.yml` -- what would you think about changing this?

Perhaps the `main.yml` could set an env variable such as "CODECOV"?

But perhaps a better, more robust solution would be to detect whether the `codecov` gem is present for this block? That way, setting the bundle `without` controls whether this block runs?